pyxem/generators/red_intensity_generator.py

```diff
@@ -0,0 +1,253 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Reduced intensity generator and associated tools.
+
+
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+
+from hyperspy.signals import Signal1D
+
+from pyxem.signals.electron_diffraction1d import ElectronDiffraction1D
+from pyxem.signals.reduced_intensity1d import ReducedIntensity1D
+
+from pyxem.components.scattering_fit_component_xtables import ScatteringFitComponentXTables
+from pyxem.components.scattering_fit_component_lobato import ScatteringFitComponentLobato
+from pyxem.utils.ri_utils import scattering_to_signal_lobato, \
+    scattering_to_signal_xtables, subtract_pattern, mask_from_pattern
+from pyxem.signals import transfer_navigation_axes
+from pyxem.signals import transfer_signal_axes
+
+scattering_factor_dictionary = {'lobato': ScatteringFitComponentLobato,
+                                'xtables': ScatteringFitComponentXTables}
+scattering_signal_dictionary = {'lobato': scattering_to_signal_lobato,
+                                'xtables': scattering_to_signal_xtables}
+
+
+class ReducedIntensityGenerator():
+    """Generates a reduced intensity 1D profile for a specified diffraction radial
```

Comment:
If this is going to be specialized to 1D lets add that to the name. If it could be generalized, then update the docstring so it doesn't say 1D.

Reply:
Yeah, I think it is better to keep it specialised to 1D, as many of the methods would be fairly different in 2D. Will update.